### PR TITLE
docs: Use storybook controls 

### PR DESCRIPTION
**Pull request by edufarre**
Replacing story knobs. Now we do have a table with the component props/API:
Screen Shot 2021-02-01 at 09 35 35
⚠️ However, beware that the `String type` properties that must be an element of a specific number of options (string[]) in order to be a valid value, do not parse to a `select` in the controls table, that why it must be set by hand. I reckon it might not be the fanciest solution, but, it does de trick.

In addition, we do have component IntelliSense as well:
Screen Shot 2021-02-01 at 09 43 14
Screen Shot 2021-02-01 at 09 41 36

### CODE REVIEW

**Review by afontcu — COMMENTED** [review]
Looks promising! Hopefully both vue-docgen-api and storybook will implement the missing pieces to make everything work flawlessly when it comes to prop `@values` 🎉

File: src/components/MaButton/MaButton.stories.js [hunk 1]
```diff
@@ -1,96 +1,43 @@
...
+  click: action('click'),
 }
 
-export const IconButton = () => {
-  const icon = select('Icons', availableIcons, availableIcons[0])
-  const iconSize = number('Icon height size in px', 18)
+const getProps = (argTypes) =>
+  Object.values(argTypes)
+    .filter((a) => a.table?.category === 'props')
+    .map((a) => a.name)
```
[the remark this answers is not in the document]
Comment (edufarre) [reply src/components/MaButton/MaButton.stories.js]: Done. I did a bit clean up into the component's story. However, I reckon that the way I do handle the `click` action, might not be the best one. I did not figure out a better approach.

**Review by sergigraciaj — APPROVED** [review]

File: src/components/MaButton/MaButton.stories.js [hunk 2]
```diff
@@ -1,96 +1,37 @@
...
     },
-  }
+    /**
+     * Tell Storybook that 'click' arg to your story should be an action
+     */
+    click: { action: 'click' },
+  },
 }
 
-export const IconButton = () => {
```
Comment (afontcu) [inline src/components/MaButton/MaButton.stories.js:66]: We lost the IconButton story, right?

File: src/components/MaButton/MaButton.vue [hunk 3]
```diff
@@ -29,30 +22,45 @@ export default {
...
+     * Sets the component displaying style
+     */
     category: {
       type: String,
       default: 'primary',
-      validator: (v) => AVAILABLE_CATEGORIES.includes(v),
+      validator: (v) =>
+        ['primary', 'secondary', 'white', 'gradient', 'no-background'].includes(
+          v
+        ),
```
Comment (afontcu) [inline src/components/MaButton/MaButton.vue:63]: This change won't be needed once this PR gets merged 😉 
https://github.com/vue-styleguidist/vue-styleguidist/pull/1051

File: src/components/MaButton/MaButton.vue [hunk 4]
```diff
@@ -29,30 +22,45 @@ export default {
   },
 
   props: {
+    /**
+     * The tag property of the component sets when the component tag
```
Comment (afontcu) [inline src/components/MaButton/MaButton.vue:26]:
```suggestion
     * Sets the HTML tag element of the component
```

**Review by cillascillas — APPROVED** [review]
It looks great in the preview! Is there any way of hiding the knobs tab now that it's empty (or any other that we don't use)?

File: src/components/MaButton/MaButton.stories.js (hunk 2 above)
Comment (edufarre) [reply src/components/MaButton/MaButton.stories.js:66]: Yep, I do believe this was on the scope of this PR. However, as far as I know, we should implement/refactor the `MaIcon` store and then use that one inside our `MaButton.stories`, thus we won't need to maintain more code than the required one
See: https://storybook.js.org/docs/vue/workflows/stories-for-multiple-components#reusing-subcomponent-stories

File: .storybook/main.js [hunk 5]
```diff
@@ -8,5 +8,6 @@ module.exports = {
     '@storybook/addon-knobs',
     '@storybook/addon-links',
     '@storybook/addon-viewport',
+    '@storybook/addon-essentials',
```
Comment (afontcu) [inline .storybook/main.js:11]: If we add the essentials addon we can then remove [the ones included in the package](https://storybook.js.org/docs/react/essentials/introduction), right?

**Comment by edufarre** [comment]
> It looks great in the preview! Is there any way of hiding the knobs tab now that it's empty (or any other that we don't use)?

I do believe we should remove the `@storybook/addon-knobs` addon

File: .storybook/main.js (hunk 5 above)
Comment (edufarre) [reply .storybook/main.js:11]: Yeah, I forgot about it. My bad.

Comment (cillascillas) [reply .storybook/main.js:11]: I'll keep an eye on that so I can remove `addons-docs` from the Docs PR!

**Comment by edufarre** [comment]
@cillascillas and @afontcu after checking the project dependencies, I noticed that we cannot delete `@storybook/addon-actions` as we still use it in other stories (we should keep it there until we refactor ALL the remaining component stories)

I deleted:
-  `@storybook/addon-links`
-  `@storybook/addon-viewport`

File: src/components/MaButton/MaButton.stories.js (hunk 2 above)
Comment (afontcu) [reply src/components/MaButton/MaButton.stories.js:66]: Not sure I 100% follow you here – what I mean is that `IconButton` story is gonne (if I'm not mistaken!) and I think is quite a common scenario that we might want to maintain documented (so we can use it as a diff tool when we improve either the Button or the Icon component).

**Pull request merged**